File: 3-leetcode/01-easy/35-search-insert-position.ts
const searchInsert = (nums: number[], target: number): number => {
  let leftIndex = 0;
  let rightIndex = nums.length - 1;

  while (rightIndex > leftIndex) {
    const middleIndex = leftIndex + Math.floor((rightIndex - leftIndex) / 2);
    const value = nums[middleIndex];

    if (value === target) {
      return middleIndex;
    } else if (value > target) {
      rightIndex = middleIndex - 1;
    } else {
      leftIndex = middleIndex + 1;
    }
  }

  return nums[leftIndex] < target ? leftIndex + 1 : leftIndex;
};

File: 3-leetcode/01-easy/cpp/35-search-insert-position.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution
{
public:
    int searchInsert(vector<int> &nums, int target)
    {
        int leftIndex = 0;
        int rightIndex = nums.size() - 1;

        while (rightIndex > leftIndex)
        {
            int middleIndex = leftIndex + (rightIndex - leftIndex) / 2;
            int value = nums[middleIndex];

            if (value == target)
            {
                return middleIndex;
            }
            else if (value > target)
            {
                rightIndex = middleIndex - 1;
            }
            else
            {
                leftIndex = middleIndex + 1;
            }
        }

        return nums[leftIndex] < target ? leftIndex + 1 : leftIndex;
    }
};
